File: 프로그래머스/unrated/181923. 수열과 구간 쿼리 2/수열과 구간 쿼리 2.js
function solution(arr, queries) {
    return queries.map(([start, end, limit]) => {
        const temp = [...arr].slice(start, end+1).filter(el => el > limit).sort((a, b) => a - b)
        return temp.length !== 0 ? temp[0] : -1
    })
}

File: 프로그래머스/unrated/181923. 수열과 구간 쿼리 2/수열과 구간 쿼리 2.py
def solution(arr, queries):
    def mapping(query):
        [s, e, k] = query
        temp = sorted(list(filter(lambda x: x > k, arr[s : e+1])))
        return temp[0] if len(temp) else -1
    
    return list(map(mapping, queries))